gravitee-management-api-rest/src/main/java/io/gravitee/management/rest/resource/auth/OAuth2AuthenticationResource.java

```diff
@@ -128,6 +127,10 @@ public Response oauth2(@Valid final Payload payload) throws IOException {
         // Step 3. Process the authenticated user.
         final String userInfo = getResponseEntityAsString(response);
         if (response.getStatus() == Response.Status.OK.getStatusCode()) {
+            //set user to Authentication Context
```

Comment:
It is better to put this authentication after verifying user data. Because, in some case, if they are no "public email" for this user, this later will not being able to authenticate. So you must move this part of code in the processUser method before the call to updateUser